tardis/adapters/sites/kubernetes.py

```diff
@@ -0,0 +1,176 @@
+        self, resource_attributes: AttributeDict
+    ) -> AttributeDict:
+        spec = client.V1DeploymentSpec(
+            replicas=1,
+            selector=client.V1LabelSelector(
+                match_labels={"app": self.machine_type_configuration.label}
+            ),
+            template=client.V1PodTemplateSpec(),
+        )
+
+        container = client.V1Container(
+            image=self.machine_type_configuration.image,
+            args=self.machine_type_configuration.args.split(","),
+            name=resource_attributes.drone_uuid,
+            resources=client.V1ResourceRequirements(
+                requests={
+                    "cpu": self.machine_meta_data.Cores,
+                    "memory": self.machine_meta_data.Memory,
+                }
+            ),
+        )
+
+        spec.template.metadata = client.V1ObjectMeta(
+            name=self.machine_type_configuration.label,
+            labels={"app": self.machine_type_configuration.label},
+        )
+
+        spec.template.spec = client.V1PodSpec(containers=[container])
+        body = client.V1Deployment(
+            metadata=client.V1ObjectMeta(name=resource_attributes.drone_uuid),
+            spec=spec,
+        )
+
+        responseTemp = await self.client.create_namespaced_deployment(
+            namespace="default", body=body
+        )
+
+        response = dict(
+            {
+                "uid": responseTemp.metadata.uid,
+                "name": responseTemp.metadata.name,
+                "type": "Progressing",
+            }
+        )
+
+        return self.handle_response(response)
+
+    async def resource_status(
+        self, resource_attributes: AttributeDict
+    ) -> AttributeDict:
+        list = await self.client.list_namespaced_deployment(namespace="default")
+
+        response = dict(
+            {
+                "uid": resource_attributes.remote_resource_uuid,
+                "name": resource_attributes.drone_uuid,
+                "type": "Deleted",
+            }
+        )
+
+        for x in list.items:
+            if x.metadata.name == resource_attributes.drone_uuid:
+                responseTemp = await self.client.read_namespaced_deployment(
+                    name=resource_attributes.drone_uuid, namespace="default"
+                )
+
+                if responseTemp.spec.replicas == 0:
+                    response = dict(
+                        {
+                            "uid": responseTemp.metadata.uid,
+                            "name": responseTemp.metadata.name,
+                            "type": "Stopped",
+                        }
+                    )
+                else:
+                    response = dict(
+                        {
+                            "uid": responseTemp.metadata.uid,
+                            "name": responseTemp.metadata.name,
+                            "type": responseTemp.status.conditions[0].type,
+                        }
+                    )
+
```

Comment:
Considering the above comment on ``response_type``/``"type": "Deleted"``: Python has a [``for``-``break``-``else`` construct](https://docs.python.org/3/reference/compound_stmts.html#for), which basically means "search `for` all then `break`, `else` do default".

Please strongly consider to refactor the ``response`` creation and search like this:
```python3
for x in list.items:
    if x.metadata.name == resource_attributes.drone_uuid:
        responseTemp = await self.client.read_namespaced_deployment(
            name=resource_attributes.drone_uuid, namespace="default"
        )
        if responseTemp.spec.replicas == 0:
            response_type = "Stopped"
        else:
            response_type = responseTemp.status.conditions[0].type
        break  # stop search once we find the resource
else:  # in case we did not find the resource
    response_type = "Deleted"
# create entire response using `response_type`
```

Reply:
changed to try: except: